src/util/cmdline.cpp

```diff
@@ -121,6 +123,14 @@ bool cmdlinet::parse(
     args = vm["input-file"].as<std::vector<std::string>>();
     verification_file = args.back();
   }
+  if (vm.count("dummymain")) // dummy main for Solidity frontend
+  {
+    sol_main = vm["dummymain"].as<std::vector<std::string>>().back();
+  }
+  if (vm.count("function")) // We need Solidity function name for pattern-based verification
+  {
+    sol_function = vm["function"].as<std::string>();
```

Comment:
Could you replace with another name so it will not conflict with the "--function" flag?

Reply:
Sorry, I think my comment in line 130 appears confusing. Please ignore it.

The intention was to re-use the "function" flag defined in `src/esbmc/options.cpp`: 

```cpp
110    {{"function",
111      boost::program_options::value<std::string>()->value_name("name"),
112      "set main function name"},
```

If we create a new flag, would that be a duplicate?

Reply:
> The intention was to re-use the "function" flag defined in src/esbmc/options.cpp:

Are you sure this is needed? Can't you use the `esbmc parseoptions` directly?

> If we create a new flag, would that be a duplicate?

I am still confused about what this does for this frontend. For the C frontend, this flag is used to define the entry point (as a `main`).

Reply:
For the Solidity frontend, this flag is used in the same way as the C frontend - it also defines the entry point. 

> Are you sure this is needed? 

Yes, it will be eventually used as the target_func in `src/solidity-frontend/pattern_check.h`.

> Can't you use the `esbmc parseoptions` directly?

If I remembered correctly, there were a bunch of patches to migrate to `boost::program_options`, which makes `parseoptions` obsolete.  (cf. [cmdline.cpp history: commits on May 22, 2021](https://github.com/esbmc/esbmc/commits/master/src/util/cmdline.cpp) ). I used `vm.count("function")` to reuse the '--function' flag for Solidity frontend.

Reply:
> If I remembered correctly, there were a bunch of patches to migrate to boost::program_options, which makes parseoptions obsolete

I mean [esbmc_parseoptions](https://github.com/esbmc/esbmc/blob/master/src/esbmc/esbmc_parseoptions.cpp), which sets the [options object](https://github.com/esbmc/esbmc/blob/master/src/esbmc/options.cpp).

Reply:
Thanks very much for the clarification! 

Given the fact that we want to re-use the '--function' flag in exactly the same way as C frontend, a new flag would become duplicate. There is no conflict to the existing C frontend because it does not break the C verification flow.  

For the time being, '--function' is used verify all functions in a Solidity smart contract one by one. I'm not 100% sure but there might be a more elegant way to verify a Solidity smart contract as a whole, and hence this line is probably open to further changes. I'll put a comment in the next patch set.

Reply:
I agree with @rafaelsamenezes that this seems odd: to make it consistent with how the other frontends work `--function` should only check one function, while when not using `--function` all function would be checked.

Reply:
Sorry, my previous comment might have caused some confusion about `--function` flag in the new Solidity frontend. 

For Solidity forntend, the `--function`flag only checks one function, which is consistent with other frontends. E.g. 
```
esbmc $CONTRACT_DIR/swc-101/MyContract_overflow.solast -M main.c --function func_overflow --compact-trace
```

> For the time being, '--function' is used verify all functions in a Solidity smart contract one by one.

What I really mean by this comment is that the above command can be wrapped in an automation script to verify all functions one by one,  something like: 
```
esbmc $CONTRACT_DIR/swc-101/MyContract_overflow.solast -M main.c --function func_1 --compact-trace
esbmc $CONTRACT_DIR/swc-101/MyContract_overflow.solast -M main.c --function func_2 --compact-trace
esbmc $CONTRACT_DIR/swc-101/MyContract_overflow.solast -M main.c --function func_3 --compact-trace
```

Apologies for the misleading comment. I was coding an automation framework to run more complex Solidity smart contract that contains multiple functions, while I was writing the previous comment. These thoughts accidently interfered and slipped into the writing. 😄

Reply:
Ok, so you shouldn't need to recreate this flag in `cmdline`, you can use from `options`.

Now I am confused about the C modules though. Could you document the full expected flow with some examples (our regression cases)?

Reply:
> Could you document the full expected flow with some examples (or regression cases)?

Sure, I'll attach a short PDF to outline the big picture as well as a few examples.